### ISSUE TITLE
scoring issues with multivalued fields since solr 6 upgrade

### ISSUE THREAD

**Issue by higrow** [sematext/solr-redis#38]
Hi,

we encounter some problems with scores on multivalued fields since the upgrade to solr 6.1.0 (we use the patched branch solr-6-1, but i can reproduce this with solr 6.0 and the master branch too)

This used to work with solr 5.4 and the solr-redis extension in Version 1.6.0

### Short description:
documents are not scored correctly when using solr-redis to query multivalued fields. 

### Details:
let's assume we have a list of users in our index with interests represented as a list of ids
```JSON
[{
	"id": "1",
	"interestIds": ["1", "2", "3"]
}, {
	"id": "2",
	"interestIds": ["1", "2", "3"]
}, {
	"id": "3",
	"interestIds": ["2", "3", "4"]
}]
```

interestIds is defined as multivalued field in the schema
```xml
<field name="interestIds" type="string" docValues="true" multiValued="true" indexed="true" required="true" stored="true"/>
```

when i query for interestIds 1 and 2 without using redis `(interestIds:1 interestIds:2)`
```bash
curl http://localhost:8983/solr/test/select?fl=score,interestIds&hl.fl=interestIds&hl=on&indent=on&q=(interestIds:1%20interestIds:2)&wt=json
```

as expected documents 1 and 2 score higher than 3

```json
"response": {
	"numFound": 3,
	"start": 0,
	"maxScore": 0.60353506,
	"docs": [{
		"interestIds": ["1", "2", "3","4"],
		"score": 0.60353506
	}, {
		"interestIds": ["1","2","3"],
		"score": 0.60353506
	}, {
		"interestIds": ["2","3","4","71"],
		"score": 0.13353139
	}]
}
```

when using redis:
```bash
redis-cli sadd interests 1 2
```

and i query with redis `({!redis command="smembers" key="interests"}interestIds)`
```bash
curl 'http://localhost:8983/solr/test/select?fl=score,interestIds&hl.fl=interestIds&hl=on&indent=on&q=(\{!redis%20command=%22smembers%22%20key=%22interests%22\}interestIds)&wt=json'
```
all documents score equal
```JSON
"response": {
	"numFound": 3,
	"start": 0,
	"maxScore": 1,
	"docs": [{
		"interestIds": ["1","2","3","4"],
		"score": 1
	}, {
		"interestIds": ["1","2","3"],
		"score": 1
	}, {
		"interestIds": ["2","3","4","71"],
		"score": 1
	}]
}
```

we see similar problems that might be related with highlighting not working on multivalued fields, but i need to build a reproducible case for that first

If i can provide additional information or help out in any way let me know. I'm happy to help

thanks in advance

**Comment by otisg** [comment]
@prog8 - this should work with Solr 6.2.  Older versions are not important.

**Comment by prog8** [comment]
When we switched to Solr 6.0 we started using TermsQuery instead of BooleanQuery. TermsQuery ignores score. It doesn't matter whether there are 2 matching terms or only 1 matching term, scoring is equal. I had to introduce new parameter ignoreScore which is by default set to false which meant BooleanQuery is used. If one wants to use more efficient query underneath (TermsQuery) he has to specify ignoreScore=true. Unluckily it is not possible to automatically enable ignoreScore=true for fq because QParser doens't know whether it is used in fq or q (https://issues.apache.org/jira/browse/SOLR-2883)
